playbooks/files/rds_preconditions.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+import sys
+import openstack
+import json
+
+
+def find_latest_version(datastores):
+    max_version = max(datastores, key=lambda x: float(x['name']))
+    return max_version['name']
+
+
+if __name__ == "__main__":
+    try:
+        if len(sys.argv) != 4:
+            print(json.dumps({"error": "check_rds_flavor.py <flavor_code>"}))
+            sys.exit(1)
+
+        # Replace placeholders with actual values
+        database_name = sys.argv[1]
+        db_versions = json.loads(sys.argv[2])
+        spec_code = sys.argv[3]
+
+        latest_version = find_latest_version(db_versions)
+
+        # Establish a connection using your OpenStack profile
+        conn = openstack.connect()
+
+        # Determine the service type and interface
+        service_type = 'rdsv3'  # Adjust this to match your service
+        interface = 'public'       # Use 'internal' or 'admin' if necessary
+
+        # Retrieve the service endpoint
+        endpoint = conn.session.get_endpoint(
+            service_type=service_type,
+            interface=interface
+        )
+
+        # Retrieve the project ID
+        project_id = conn.current_project_id
+
+        # Retrieve region name
+        region_name = conn.config.region_name
+
+        # Construct the URL
+        url1 = (
+            f"{endpoint}/storage-type/{database_name}"
+            f"?version_name={latest_version}"
+        )
+
+        # Make the GET request using the authenticated session
+        response = conn.session.get(url1)
+        data1 = response.json()
+
+        # Construct the URL
+        url2 = (
+            f"{endpoint}/flavors/{database_name}?spec_code={spec_code}"
```

Comment:
why this construction?
we have flavors integrated in python SDK so you should use it natively

Reply:
SDK does not provide all necessary response parameters. Especially "group_type".
I have opened issue to fix SDK https://github.com/opentelekomcloud/python-otcextensions/issues/470 .
But I need to make this fix in place as soon as possible as currently it is generating alerts for no eu-de regions OTCDB-7525 .

Reply:
fixed